### DISCUSSION TITLE
Retornar HTML gerado ou JSON?

### DISCUSSION THREAD

**Discussion by paulinhoerry** [frontendbr/forum#799] [Perguntas e Respostas]
Então galera, Atualmente trabalho numa empresa onde a maior parte da lógica do sistema fica no backend que é em Zend. O frontend fica apenas encarregado de imprimir as views e algumas manipulações na UI, como validação, máscaras e etc. Andei pesquisando e em alguns lugares diz que isso é uma má prática, já outros dizem que não.

No que diz respeito a performance, teria um impacto muito grande mudar esses retornos para um JSON e o frontend montar toda a view ou a diferença seria pouca?

**Comment by bernardodiasc** [comment]
Isso depende bastante, na prática não tem nenhum problema na sua abordagem se soluciona seu problema. SPA tem benefícios mas tambem tem probemas, existem os casos que são indicados, se for o seu, pode ser que seja boa ideia migrar, mas realmente depende. Voce consegue identificar problemas que voce possa estar tendo ou possa vir a ter com o tempo na sua abordagem?

**Comment by paulinhoerry** [comment]
@bernardodiasc primeiramente, obrigado pela atenção.

então, atualmente o nosso maior problema tá sendo justamente a performance. Algumas páginas com tabela demoram um pouco pra abrir e fica cansativo olhar o loading por 4s em média. 
Uma das cogitações seria o que mencionei acima, descarregar um pouco a responsabilidade do backend e jogar no frontend, mas até o entanto ainda não achei abordagens plausíveis para essa mudança.

**Comment by bernardodiasc** [comment]
Hmm entendi, tem que ver se vale a pena mesmo, se for soh por esse motivo tvz não tenha tanto ganho assim. Mas a experiência de usar um webapp costuma ser melhor que um website mesmo.

**Comment by CinthiaQueiro** [comment]
Boa noite.

Em teoria a manipulação de dados ocorre mais rapidamente no back end do que
no front. Apesar de não ser uma boa prática.  O que você precisa
identificar ai é se a demora está ocorrendo devido a montagem do Json ou
dos selects no banco de dados mesmo. Um banco de dados mal estruturado faz
com que o desempenho caia muito. Façam testes e identifiquem se é o caso.


Em 7 de nov de 2017 20:32, "Bernardo Dias" <notifications@github.com>
escreveu:

**Comment by laurindo** [comment]
Paulo Rodrigues, como a Cinthia falou, pode ser que o problema nem seja na
renderização do HTML. Pode ser um conjunto de fatores.
Consulta ao banco pode ser a demora, se for, será que indexar alguma coluna
resolve?
Se a quantidade de dados pra renderizar cada página pode ser grande?
Se vc sempre está carregando o HTML inteiro.
Possui algum sistema de cache pra determinadas chamadas que quase sempre
não mudam?
Tem vários fatores que podem melhorar o carregamento.

Link interessante, talvez ajude com algumas dicas :
https://testmysite.thinkwithgoogle.com/intl/pt-br

E acho que o ideal é separar entre back e front.

**Comment by CinthiaQueiro** [comment]
Perfeito Daniel.

Em 17 de nov de 2017 23:41, "Daniel Laurindo" <notifications@github.com>
escreveu:

> Paulo Rodrigues, como a Cinthia falou, pode ser que o problema nem seja na
> renderização do HTML. Pode ser um conjunto de fatores.
> Consulta ao banco pode ser a demora, se for, será que indexar alguma coluna
> resolve?
> Se a quantidade de dados pra renderizar cada página pode ser grande?
> Se vc sempre está carregando o HTML inteiro.
> Possui algum sistema de cache pra determinadas chamadas que quase sempre
> não mudam?
> Tem vários fatores que podem melhorar o carregamento.
>
> Link interessante, talvez ajude com algumas dicas :
> https://testmysite.thinkwithgoogle.com/intl/pt-br
>
> E acho que o ideal é separar entre back e front.
>
